src/main/java/com/exasol/exasoltestsetup/WaitHelper.java

```diff
@@ -29,10 +31,19 @@ public static void waitFor(final long milliSeconds) {
      * This method checks the predicate every 200 ms.
      * </p>
      * 
-     * @param predicate function that tells if waiting should be over
+     * @param predicate       function that tells if waiting should be over
+     * @param timeoutAfterSec maximum number of seconds to wait
+     * @param jobDescription  name of the condition to wait on. Used in the timeout exception.
      */
-    public static void waitUntil(final BooleanSupplier predicate) {
+    public static void waitUntil(final BooleanSupplier predicate, final int timeoutAfterSec,
+            final String jobDescription) {
+        final long start = System.currentTimeMillis();
         while (!predicate.getAsBoolean()) {
+            final long elapsed = System.currentTimeMillis() - start;
+            if (elapsed > timeoutAfterSec * 1000L) {
```

Comment:
I usually prefer to use TimeUnit for this kind of conversion...

Reply:
:+1: